frontend/src/components/NotFound.vue

```diff
@@ -1,17 +1,27 @@
 <template>
   <div class="column has-text-centered is-three-fifths-desktop is-three-quarters-tablet is-fullwidth-mobile">
     <div class="box has-background-light content">
-      <template v-if="type">
-        <p class="title is-size-5">
-          <span class="is-capitalized">{{ type }}&nbsp;</span> "&nbsp;{{ componentId }}&nbsp;" &nbsp;not found
-        </p>
-        <template v-if="type === 'model'">
-          <p>Visit the Explore page to select one of the integrated models</p>
+      <template v-if="componentId">
+        <template v-if="type">
+          <p class="title is-size-5">
+            <span class="is-capitalized">{{ type }}&nbsp;</span> "&nbsp;{{ componentId }}&nbsp;" &nbsp;not found
+          </p>
+          <template v-if="type === 'model'">
+            <p>Visit the Explore page to select one of the integrated models</p>
+          </template>
+          <template v-else>
+            <p>
+              Probably there is a typo in the {{ type }} identifier in the URL
+              <br>Use the search bar above to find other {{ type }}s
+            </p>
+          </template>
         </template>
         <template v-else>
+          <p class="title is-size-5">
+            <span class="is-capitalized">{{ componentId }}</span>&nbsp;" &nbsp;not found
```

Comment:
Is the space between quotation mark and the componentId added by purpose? When you look at the rendered "not found" page, it seems there are trailing white spaces around the ID.

Reply:
Thanks for this comment. Since this is from an outdated commit, I will refer to [this line](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/544/files#diff-3a5b53f2754c9efb675a20e09d7f9cc439c90d32a80736950ba1ea75e23c6126R6) in the diff: the whitespaces are unchanged in this PR. I imagine that the whitespaces were initially added to avoid the confusion if the quotation marks could be a part of the ID. I did notice that the lines in this file are not indented properly though so I will try to push a fix for that soon.

Reply:
Yes that's the explanation for the extra space. Any other ideas to do this? Code markup?

Reply:
I support using the `code` tag for this.

Reply:
I agree. using `code` tag might be better than adding extra white space.

Reply:
Let's go for `that`!

Reply:
It looks really neat, it's so much better than the trailing spaces. Thanks for bringing this up @nanjiangshu and for the fix @e0.